docs/tfengine/dev.md

```diff
@@ -0,0 +1,47 @@
+# Terraform Engine Development Guide
+
+## New Resource
+
+1. Verify the resource does not already exist in the resources
+  [list](./schemas/resources.md).
+
+1. If it does not exist, add a new
+  [component](../../templates/tfengine/components/resources).
+
+  See
+  [Storage Buckets](../../templates/tfengine/components/resources/storage_buckets)
+  for an example to copy.
+
+1. When implementing the template, prefer to use
+  [Cloud Foundation Toolkit](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/docs/terraform.md)
+  modules. If none is available, use a resource directly from the
```

Comment:
Nit: Add "or consider writing a new CFT module if you anticipate it would be useful to the community."